server/routerlicious/packages/services-shared/src/configDumper.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fastRedact from "fast-redact";
+import { ILogger } from "@fluidframework/server-services-core";
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+const errorSanitizationMessage = "FluidREDACTED";
+
+export class ConfigDumper {
+	private readonly config: Record<string, any>;
+	private readonly secretNamesToRedactInConfigDump = [
+		"mongo.globalDbEndpoint",
+		"mongo.operationsDbEndpoint",
+		"redis.pass",
+		"redisForTenantCache.pass",
+		"redis2.pass",
+		"redisForThrottling.pass",
+	];
+	private readonly logger: ILogger | undefined;
+
+	// Check library for issues/malware
+	constructor(
+		config: Record<string, any>,
+		logger: ILogger | undefined,
+		secretNamesToRedactInConfigDump?: string[],
+	) {
+		this.config = JSON.parse(JSON.stringify(config));
+		if (secretNamesToRedactInConfigDump !== undefined) {
+			this.secretNamesToRedactInConfigDump = this.secretNamesToRedactInConfigDump.concat(
```

Comment:
for my understanding, what happens when there are duplicates in this list? Hope it does not throw an error.

Reply:
I tried with duplicates on my local env. The redacting worked as expected with no errors. Maybe I can add a unit test for this scenario?